### PR TITLE
Cassandra functioning charm

### PR DESCRIPTION
**Pull request by dstathis**
Cassandra functioning charm

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -5,8 +5,11 @@ description: |
 series:
   - kubernetes
 provides:
-  database:
+  cql:
```
Comment (balbirthomas) [inline metadata.yaml:8]: Is "cql" supposed to mean some thing specific ?

File: src/charm.py [hunk 2]
```diff
@@ -50,20 +63,56 @@ def build_pod_spec(self):
...
+                        "CASSANDRA_SEEDS": self.seeds(),
+                        "CASSANDRA_LISTEN_ADDRESS": "listen-addr",
+                    },
                 }
             ],
         }
         return spec
 
+    def seeds(self):
+        seeds = UNIT_ADDRESS.format(self.meta.name, 0, self.meta.name, self.model.name)
```
Comment (balbirthomas) [inline src/charm.py:91]: Maybe I am missing something, but it is not clear to me why you don't do something like
```
for i in range(min(3, self.num_units())):
     seeds.append(UNIT_ADDRESS.format(....))
```
and convert to a tuple if you must.

**Review by balbirthomas — APPROVED** [review]

File: metadata.yaml (hunk 1 above)
Comment (dstathis) [reply metadata.yaml:8]: cassandra query language

File: src/charm.py (hunk 2 above)
Comment (dstathis) [reply src/charm.py:91]: I would have to be that followed by a ','.join. That might still be cleaner though

**Review by Abuelodelanada — APPROVED** [review]
Looks good to me.
As Balbir said may be the seeds() method may be cleaner

**Pull request merged**